test/worm_test.gleam

```diff
@@ -41,3 +43,45 @@ pub fn readme_example_test() {
   |> regex.scan(with: initial)
   |> should.equal([regex.Match("Hello", [])])
 }
+
+fn cached_side_effect(filepath: String, value: String) -> String {
+  use <- worm.persist()
+
+  let assert Ok(_) = "Hello world!" |> simplifile.append(to: filepath)
+
+  value
+}
```

Comment:
As much as you can write this, the intention is that this is invalid code. We can't enforce it, but using `worm.persist()` in functions that take arguments leads to calls that don't make sense. In particular, it's documented that the callback should be *deterministic* and *pure*. Neither of these requirements are met by this function.

That said, these tests are useful examples of what not to do and why. Can you move these tests to a new test file (lets say `test/worm/bad_examples.gleam`) so that we can keep the counterexamples contained somewhere that is obviously examples of things not to do?

Reply:
This is what I get for just skimming the docs. I agree that this advice makes sense generally though it's unfortunate that it can't be enforced statically.

Is it really still useful to have these tests at all in that case? Maybe they would be better as documentation somewhere or should just be excluded entirely.

Reply:
I wouldn't have written these tests myself, but they're written now and I think it's probably a good idea to make sure that the things we expect to break do in fact continue to break. It's important to test happy paths as well as breakages.

I wouldn't want to include examples of bad code directly in the documentation, but a link off to this in the main module documentation would make sense, in the limitations sections.

Docstrings for each test about why they don't work in a useful way would also be good.